src/user-search/search.ts

```diff
@@ -0,0 +1,42 @@
+import {JsonRpcAuthMethodContext as JCtx} from '@steemit/koa-jsonrpc'
+import * as config from 'config'
+import * as Koa from 'koa'
+import {users as gdprList } from '../../user-data/lists/gdpr/users'
+import {Context} from '../server'
+import {matchPrefix} from './indexes'
+import {logger} from '../logger'
+
+const ADMIN_ACCOUNT = config.get('admin_role')
+
+interface JCtx2 extends JCtx {
+    ctx: Context
+}
+
+export async function getAccount(this: JCtx2,
+                                 account: string,
+                                 contextAccount: string) {
+    this.assert(this.account === ADMIN_ACCOUNT, 'Unauthorized')
+    const client: any = this.ctx.cacheClient
+    if (gdprList.has(account)) {
+        return []
+    }
+    logger.info(`getAccount account:${account} contextAccount:${contextAccount}`)
+    return await client.loadAccountJSON(account, contextAccount)
+}
+
+export async function autocompleteAccount(this: JCtx2,
+                                          account: string,
+                                          accountSubstring: string
+                                         ) {
+    this.assert(this.account === ADMIN_ACCOUNT, 'Unauthorized')
+    const client: any = this.ctx.cacheClient
+    const userAccount = client.loadAccount(account)
+    const globalAccountNames = matchPrefix(this.ctx.userAccountTrie, accountSubstring)
+    const friendAccountNames = new Set(globalAccountNames.filter((x) => userAccount.following.has(x)))
```

Comment:
would it make sense to store this data in postgres?

Reply:
Yes, with an upper bound of `num_users * (num_users - 1)/2`  we can use as much memory as we choose to afford. I have a configurable default TTL of 10 minutes for everything in the memory. Each kind of item in the cache can have its own TTL too, with `UserContext` being a good example of something that could probably use a lower TTL.